src/Fieldset/index.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import newId from '../utils/newId';
+import styles from './Fieldset.scss';
+import ValidationMessage from '../ValidationMessage/index';
+import Variant from '../utils/constants';
+
+const inputProps = {
+  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+
+  children: PropTypes.node,
+  className: PropTypes.string,
+  id: PropTypes.string,
+  isValid: PropTypes.bool,
+  invalidMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  variant: PropTypes.shape({
+    status: PropTypes.string,
```

Comment:
Should this be `PropTypes.oneOf(Variant.status)`? We probably want to have `prop` warnings if somebody puts in `foobar` or something?